### PR TITLE
tools: Use own go.mod

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
This reduces the number of deps that will be fetched for building ksm

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,7 +150,8 @@ scripts/vendor: scripts/jsonnetfile.json scripts/jsonnetfile.lock.json
 
 install-tools:
 	@echo Installing tools from tools.go
-	@cat tools/tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go install %
+	@cd tools && cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go get %
```
Comment (dgrisonnet) [inline Makefile:153]: why is this needed?

File: Makefile [hunk 2]
```diff
@@ -150,7 +150,8 @@ scripts/vendor: scripts/jsonnetfile.json scripts/jsonnetfile.lock.json
...
+	@cd tools && cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go install %
```
Comment (dgrisonnet) [inline Makefile:154]: instead of changing directory we could use `go install -mod=readonly -modfile=tools/go.mod`

File: Makefile (hunk 1 above)
Comment (mrueg) [reply Makefile:153]: The go.mod in the parent directory will record dependencies if we are not changing to a different directory.

Comment (dgrisonnet) [reply Makefile]: It shouldn't be the case if we specify "-mod=readonly" as commented below

Comment (mrueg) [reply Makefile]: Yep, that way it works. thanks for the suggestion!

Comment (dgrisonnet) [reply Makefile]: You're welcome!

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve
